### PR TITLE
Supports simply use of regular expression

### PR DESCRIPTION
**Pull request by shuklajay117**
Changes:
- Expose a method to add named regular expression
- Replace registered expression from url if used any.
- Updated tests

@amitguptagwl  Please review

### CODE REVIEW

File: src/letsRoute.js [hunk 1]
```diff
@@ -36,6 +55,18 @@ Anumargak.prototype._on = function (method, url, fn) {
         }
     }
 
+    var namedExpressionMatches = getNamedExpressionMatches(url, namedExpressionRegexStr);
```
Comment (amitguptagwl) [inline src/letsRoute.js]: We can extract this logic to a function.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "anumargak",
-  "version": "1.4.1",
+  "version": "1.5.0",
```
Comment (amitguptagwl) [inline package.json]: Please revert this change. I prefer to update the package related information just before the release.

File: src/letsRoute.js (hunk 1 above)
Comment (shuklajay117) [reply src/letsRoute.js]: I didn't get. Do you mean to say move to utils?

Comment (amitguptagwl) [reply src/letsRoute.js]: No, just extract the logic to another function in the same class. Eg `function replaceNamedExpressions()`. So that the size of `Anumargak.prototype._on()` will not increase.

File: package.json (hunk 2 above)
Comment (amitguptagwl) [reply package.json]: Hmm! I can see. Git is showing my many changes as a part of this PR. Ignore it for the time being. Once all the changes are done, we'll squash the changes. I hope we should not see any issue that time.

File: src/letsRoute.js [hunk 3]
```diff
@@ -43,7 +43,24 @@ Anumargak.prototype.on = function (method, url, fn) {
 var wildcardRegexStr = "\\/([^\\/:]*)\\*";
 var paramRegexStr = ":([^\\/\\-\\(]+)-?(\\(.*?\\))?";
 var enumRegexStr = ":([^\\/\\-\\(]+)-?(\\(([\\w\\|]+)\\))";
-var namedExpressionRegexStr = "\\(:(.*?):\\)";
+
+function replaceNamedExpression(url, _this) {
```
Comment (amitguptagwl) [inline src/letsRoute.js]: You may change it to `Anumargak.prototype.replaceNamedExpression()`. So you'll not need to pass `this`.

File: src/letsRoute.js [hunk 4]
```diff
@@ -55,17 +72,7 @@ Anumargak.prototype._on = function (method, url, fn) {
...
-        for (var i = 0; i < namedExpressionMatches.length; i++) {
-            var matchedNamedKey = namedExpressionMatches[i][1];
-            if (this.namedExpression[matchedNamedKey]) {
-                url = url.replace(":" + matchedNamedKey + ":", this.namedExpression[matchedNamedKey]);
-            } else {
-                throw Error("Usage of named expression in url as " + matchedNamedKey + ". Define it using addNamedExpression method before using in URLs.");
-            }
-        }
-    }
+    url = replaceNamedExpression(url, this);
```
Comment (amitguptagwl) [inline src/letsRoute.js]: this line would be changed to `url = this.replaceNamedExpression(url);`

Comment (shuklajay117) [reply src/letsRoute.js]: I think  better to use bind as this method should not be exposed I guess. Let me do that and you review it.

**Comment by shuklajay117** [comment]
I'm struggling with squash commits which I've done yesterday and earlier. Can you guide me on this?

Hopefully, I'm able to squash all commits which I've made just now and also able to remove your updates from this PR.

**Comment by amitguptagwl** [comment]
I checked your commits. So ideally you should not see any issue. However you can pull my changes and rebase your changes

```bash
git pull --rebase origin master
```

Add remote pointing to this repo and use this name instead of origin in above command.

**Comment by shuklajay117** [comment]
I did as you suggested and still I see older commits. Also, I see 2 commits in my local branch where I see one of that is yours.

**Comment by shuklajay117** [comment]
Finally able to manage. Thanks for your help.

**Comment by amitguptagwl** [comment]
Awesome
